src/compiler/checker.ts

```diff
@@ -14133,19 +14178,100 @@ namespace ts {
+                        if (checkBody) {
+                            const globalAugmentation = isNameOfGlobalAugmentation(<LiteralExpression>node.name);
+                            // body of ambient external module is always a module block
+                            for (const statement of (<ModuleBlock>node.body).statements) {
+                                checkBodyOfModuleAugmentation(statement, globalAugmentation);
+                            }
+                        }
                     }
-                    if (isExternalModuleNameRelative(node.name.text)) {
-                        error(node.name, Diagnostics.Ambient_module_declaration_cannot_specify_relative_module_name);
+                    else if (isGlobalSourceFile(node.parent)) {
+                        if (isExternalModuleNameRelative(node.name.text)) {
+                            error(node.name, Diagnostics.Ambient_module_declaration_cannot_specify_relative_module_name);
+                        }
+                    }
+                    else {
+                        // Node is not an augmentation and is not located on the script level.
+                        // This means that this is declaration of ambient module that is located in other module or namespace which is prohibited.
+                        error(node.name, Diagnostics.Ambient_modules_cannot_be_nested_in_other_modules_or_namespaces);
                     }
                 }
             }
             checkSourceElement(node.body);
         }
 
+        function isNameOfGlobalAugmentation(node: LiteralExpression): boolean {
+            // global augmentation
+            // TODO: fix to use 'declare global' syntax.
+            return node.text === "/";
+        }
+
+        function checkBodyOfModuleAugmentation(node: Node, isGlobalAugmentation: boolean): void {
+            switch (node.kind) {
+                case SyntaxKind.VariableStatement:
+                    // error each individual name in variable statement instead of marking the entire variable statement
+                    for (const decl of (<VariableStatement>node).declarationList.declarations) {
+                        if (isBindingPattern(decl.name)) {
+                            for (const el of (<BindingPattern>decl.name).elements) {
+                                // mark individual names in binding pattern
+                                checkBodyOfModuleAugmentation(el, isGlobalAugmentation);
+                            }
+                        }
+                        else {
+                            checkBodyOfModuleAugmentation(decl, isGlobalAugmentation);
+                        }
+                    }
+                    break;
+                case SyntaxKind.ExportDeclaration:
+                    grammarErrorOnFirstToken(node, Diagnostics.Exports_are_not_permitted_in_module_augmentations);

```

Comment:
This is a little misleading. Maybe the message should be "Exports must occur at declaration sites in a module augmentation."